src/shared_realm.hpp

```diff
@@ -188,7 +192,7 @@ class Realm : public std::enable_shared_from_this<Realm> {
         ShouldCompactOnLaunchFunction should_compact_on_launch_function;
 
         bool immutable() const { return schema_mode == SchemaMode::Immutable; }
-        bool read_only() const { return schema_mode == SchemaMode::ReadOnly; }
+        bool read_only() const { return schema_mode == SchemaMode::ReadOnlyAlternative; }
```

Comment:
There's a similar issue with this being named `read_only()`, since any existing calls to this really should be changed to `immutable()` to preserve the existing semantics.